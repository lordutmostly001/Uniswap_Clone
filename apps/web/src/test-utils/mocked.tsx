import type { MockedFunction } from 'vitest'
import { vi } from 'vitest'

/**
 * Casts the passed function as a vitest Mock.
 * Use this in combination with vi.mock() to safely access functions from mocked modules.
 *
 * @example
 *
 *  import { useExample } from 'example'
 *  vi.mock('example', () => ({ useExample: vi.fn() }))
 *  beforeEach(() => {
 *    asMock(useExample).mockImplementation(() => ...)
 *  })
 */
// vitest expects mocks to be coerced (eg fn as a Vitest MockedFunction<T>), but this is not ergonomic when using ASI.
// Instead, we use this utility function to improve readability and add a check to ensure the function is a mock.
export function mocked<T extends (...args: any) => any>(fn: T) {
  const isMock = typeof vi !== 'undefined' && vi.isMockFunction(fn)

  if (!isMock) {
    throw new Error('fn is not a mock')
  }

  return fn as MockedFunction<T>
}
