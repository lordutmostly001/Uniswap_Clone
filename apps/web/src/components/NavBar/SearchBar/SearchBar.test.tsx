import { SearchBar } from 'components/NavBar/SearchBar'
import mockMediaSize from 'test-utils/mockMediaSize'
import { render, screen } from 'test-utils/render'

vi.mock('tamagui', async () => {
  const actual = await vi.importActual('tamagui')
  return {
    ...actual,
    useMedia: vi.fn(),
  }
})

vi.mock('uniswap/src/components/modals/ScrollLock', () => ({
  useUpdateScrollLock: vi.fn(),
}))

describe('disable nft on searchbar', () => {
  beforeEach(() => {
    mockMediaSize('xxl')
  })

  it('should render searchbar on larger screen', () => {
    const { container } = render(<SearchBar />)
    expect(container).toMatchSnapshot()
    const input = screen.getByTestId('nav-search-input')
    expect(input).toBeInTheDocument()
  })
})
