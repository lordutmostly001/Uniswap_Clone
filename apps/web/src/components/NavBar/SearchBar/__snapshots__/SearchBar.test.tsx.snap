// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`disable nft on searchbar > should render searchbar on larger screen 1`] = `
.c1 {
  background-color: rgba(34,34,34,0.05);
  color: rgba(19,19,19,0.63);
  padding: 0px 8px;
  width: 20px;
  height: 20px;
  border-radius: 4px;
  font-size: 12px;
  font-weight: 535;
  line-height: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0.6;
  -webkit-backdrop-filter: blur(60px);
  backdrop-filter: blur(60px);
}

.c0 {
  width: 20px;
  height: 20px;
}

<div>
  <span
    class=""
    style="display: contents;"
  >
    <span
      class="font_unset _display-inline _boxSizing-border-box _wordWrap-break-word _whiteSpace-pre-wrap _position-absolute _width-1px _height-1px _mt--1px _mr--1px _mb--1px _ml--1px _zIndex--10000 _overflowX-hidden _overflowY-hidden _opacity-1e-8 _pointerEvents-none"
    >
      <h2
        class="is_DialogTitle font_heading _display-inline _boxSizing-border-box _wordWrap-break-word _userSelect-auto _color-color _whiteSpace-normal _fontFamily-f-family _mt-0px _mr-0px _mb-0px _ml-0px"
        data-disable-theme="true"
        id="title-:r4:"
        role="heading"
      />
    </span>
    <div
      class="is_TouchableArea _transform-0active-scale11281 _opacity-0active-0--75 _containerType-_platformweb_normal _transform-0focus-visible-scaleX0--98sc535634366 _outlineWidth-0focus-visible-1px _outlineOffset-0focus-visible-1px _outlineStyle-0focus-visible-solid _outlineColor-0focus-visible-neutral3 _display-flex _alignItems-stretch _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column _borderTopLeftRadius-t-radius-ro291586453 _borderTopRightRadius-t-radius-ro291586453 _borderBottomRightRadius-t-radius-ro291586453 _borderBottomLeftRadius-t-radius-ro291586453 _backgroundColor-transparent _cursor-pointer _width-340px t_group_true"
      data-testid="nav-search-input"
      role="button"
      tabindex="0"
    >
      <div
        class="_backgroundColor-0hover-surface1Hov3121676 _color-_grouptrue-hover_accent3Hove112785 _backgroundColor-_grouptrue-hover_surface2Hov3121676 _display-flex _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-row _backgroundColor-surface2 _borderTopWidth-1px _borderRightWidth-1px _borderBottomWidth-1px _borderLeftWidth-1px _borderTopColor-surface3 _borderRightColor-surface3 _borderBottomColor-surface3 _borderLeftColor-surface3 _pt-t-space-spa94665593 _pb-t-space-spa94665593 _pr-t-space-spa1360334076 _pl-t-space-spa1360334076 _borderTopLeftRadius-t-radius-ro291586424 _borderTopRightRadius-t-radius-ro291586424 _borderBottomRightRadius-t-radius-ro291586424 _borderBottomLeftRadius-t-radius-ro291586424 _height-40px _alignItems-center _justifyContent-space-betwe3241 _color-accent3 _borderBottomStyle-solid _borderTopStyle-solid _borderLeftStyle-solid _borderRightStyle-solid"
      >
        <div
          class="_display-flex _alignItems-stretch _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-row _gap-t-space-spa1360334080"
        >
          <div
            class="c0"
            data-cy="nav-search-icon"
          >
            <svg
              fill="none"
              height="20px"
              stroke="rgba(19, 19, 19, 0.63)"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="11"
                cy="11"
                r="8"
              />
              <line
                x1="21"
                x2="16.65"
                y1="21"
                y2="16.65"
              />
            </svg>
          </div>
          <span
            class="font_body _display-inline _boxSizing-border-box _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _color-neutral2 _fontFamily-f-family _wordWrap-break-word _fontSize-f-size-medi3736 _lineHeight-f-lineHeigh507465454 _fontWeight-f-weight-bo3548 _textAlign-left"
            data-disable-theme="true"
          >
            Search tokens
          </span>
        </div>
        <div
          class="c1"
        >
          /
        </div>
      </div>
    </div>
    <div
      style="display: contents;"
    />
  </span>
</div>
`;
